blocks/main_loop.py

```diff
@@ -73,13 +77,14 @@ def run(self):
                 self._run_extensions('before_training')
                 self.algorithm.initialize()
                 self.status._training_started = True
+            else:
+                self._run_extensions('on_resumption')
             while self._run_epoch():
                 pass
-        except KeyboardInterrupt:
-            self._run_extensions('on_interrupt')
         except TrainingFinish:
             self.log.current_row.training_finished = True
         finally:
+            signal.signal(signal.SIGINT, default_action)
```

Comment:
`default_action` does not get called, it is just set as a new handler. Therefore `after_training` always gets called, and this is where saving is actually done.

There was indeed certain inconsistency: changing back the default handling scheme should be done last, not before `after_training` is handled. I fixed that plus added a bit more docs, see #143.